lib/downloader.rb

```diff
@@ -0,0 +1,35 @@
+# Defines how chromebrew downloads files.
+require_relative 'color'
+
+def notimplemented(package)
+  puts "#{package} has not been implemented as a downloader for Chromebrew yet. Defaulting to curl.".orange
+  @_crew_downloader = "curl -#LC -"
+  def download_file(url, filename)
+    system "#{@_crew_downloader} \'#{url}\' --output #{filename}"
+  end
+end
+
+case ENV["CREW_DOWNLOADER"]
+when 'aria2', 'aria2c'
+  def download_file(url, filename)
+    system "aria2c -c \'#{url}\' --out #{filename}"
+  end
+when 'curl', '', nil
+  def download_file(url, filename)
+    system "curl -#LC - \'#{url}\' --output #{filename}"
+  end
+when 'wget'
+  def download_file(url, filename)
+    system "wget -c \'#{url}\' --output-document #{filename}"
+  end
+when 'system curl', 'system'
```

Comment:
Combine this with the `curl` option?
```ruby
when 'curl', '/usr/bin/curl', ''
  cmd = ( ENV['CREW_DOWNLOADER'] || 'curl' )
  system "#{cmd} -#LC - '#{url}' -o #{filename}"
```

Reply:
We may also unset the LD_LIBRARY_PATH to get rid of segfails when libs in CREW_PREFIX are corrupted

Reply:
I can see why that would make sense DRY wise, however it doesn't work the `system` or `system curl` parameters.